app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/StripePaymentActivity.java

```diff
@@ -138,18 +150,26 @@ private void requestTicket() {
 
         setPurchaseRequestLoading();
         try {
-            Ticket ticket = TUMCabeClient
+            TUMCabeClient
                     .getInstance(StripePaymentActivity.this)
-                    .purchaseTicketStripe(1,
+                    .purchaseTicketStripe(StripePaymentActivity.this, ticketHistory,
                             paymentSession.getPaymentSessionData().getSelectedPaymentMethodId(),
                             getUserMailAddress(),
-                            cardholder);
-            //TODO: Add Ticket to local database and jump to ShowTicketActivity
-            EventsController ec = new EventsController(StripePaymentActivity.this);
-            List<Ticket> ticketList = new ArrayList<>();
-            ticketList.add(ticket);
-            ec.addTickets(ticketList);
-            finishLoadingPurchaseRequestSuccess(ticket);
+                            cardholder, new Callback<Ticket>() {
+                                @Override
+                                public void onResponse(Call<Ticket> call, Response<Ticket> response) {
+                                    EventsController ec = new EventsController(StripePaymentActivity.this);
+                                    List<Ticket> ticketList = new ArrayList<>();
```

Comment:
Wut? Why is this empty? Please add a `// TODO`

Reply:
By "empty" you meant the OnFailure method right?
added an error message there.